### PR TITLE
extensions: add core20 support to kde-neon

### PR DESCRIPTION
**Pull request by sergiusens**
Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

**Comment by jriddell** [comment]
I also tend to add these plugs by default:
home
network
network-bind
unity7
pulseaudio
cups-control
removable-media

it might make sense for this extension to add them

**Comment by ogra1** [comment]
> pulseaudio

note that pulseaudio has long been deprecated for ```audio-playback``` and ```audio-record``` you should probably update to use them instead ...

**Comment by sergiusens** [comment]
I will address each of these

> I also tend to add these plugs by default:
> home

We should look into using portals instead, if a particular snap still needs it, they are not prevented from adding it.

> network
> network-bind

Not all snaps need network, this should also be a per snap setting.

> unity7

I don't believe this is needed, replaced with desktop-legacy (@anonymouse64 can you confirm?)

> pulseaudio

We should use audio-record and audio-playback instead, also per snap.

> cups-control

I would make this per snap too, but if we end up adding it, maybe we should for the gnome ones too (@kenvandine what are your thoughts on this?). I also believe cups is moving to some for of portal.

> removable-media

Similar to home, would make sense to use portals by default if possible here.

**Comment by anonymouse64** [comment]
> I don't believe this is needed, replaced with desktop-legacy (@anonymouse64 can you confirm?)

yes unity7 is deprecated in favor of desktop-legacy, though of course if you can get away with not using desktop-legacy that's also ideal, but desktop-legacy is the "newer legacy" interface to use

> cups-control

snaps wanting to print things should probably use `cups` and then connect to the upcoming cups snap (though I will let Ken (and likely Till at the end of the ping chain) comment on when this will be available), but the advantage is that `cups` will auto-connect, while cups-control does not auto-connect by default. `cups` interface does need a rather recent snapd though, 2.47

File: snapcraft/plugins/v2/cmake.py [hunk 1]
```diff
@@ -84,6 +90,7 @@ def _get_cmake_configure_command(self) -> str:
             '"${SNAPCRAFT_PART_SRC_WORK}"',
             "-G",
             f'"{self.options.cmake_generator}"',
+            "${SNAPCRAFT_CMAKE_ARGS}",
```
Comment (cjp256) [inline snapcraft/plugins/v2/cmake.py:93]: Can't we just use a `cmake-parameters` in the extension?

Comment (sergiusens) [reply snapcraft/plugins/v2/cmake.py:93]: The strict goal for extensions is to always extend snapcraft.yaml, if we can eventually pull it off, would be good

**Review by cjp256 — APPROVED** [review]

**Pull request merged**